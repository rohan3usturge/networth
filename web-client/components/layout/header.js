import Head from "next/head";
import Link from "next/link";

const DocsLink = process.env.NEXT_PUBLIC_DOCS_URL;

const Header = ({ locale }) => (
  <div>
    <Head>
      <title>Net Worth Calculator</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <a className="navbar-brand" href="#">
        <span className="h4">
          NET WORTH <em>calc</em>
        </span>
      </a>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>

      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav me-auto mb-2 mb-lg-0">
          <li
            className={`nav-item ${
              locale && locale.toLowerCase() == "en-ca" ? "active" : ""
            }`}
          >
            <Link href="/" locale="en-ca">
              <a className="nav-link">English</a>
            </Link>
          </li>
          <li
            className={`nav-item ${
              locale && locale.toLowerCase() == "fr-ca" ? "active" : ""
            }`}
          >
            <Link href="/" locale="fr-ca">
              <a className="nav-link">fran√ßais</a>
            </Link>
          </li>
          <li className="nav-item">
            <a className="nav-link" target="_blank" href={DocsLink}>
              Docs
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
);

export { Header };
