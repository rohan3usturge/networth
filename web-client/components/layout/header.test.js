import { Header } from "./header";
import { shallow } from "enzyme";
import { render, fireEvent, screen } from "@testing-library/react";

describe("Header", () => {
  it("should render properly and match snapshot", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render properly and match snapshot for french", () => {
    const wrapper = shallow(<Header locale="fr-ca" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render properly and match snapshot for english", () => {
    const wrapper = shallow(<Header locale="en-ca" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render properly and match snapshot", () => {
    render(<Header locale="fr-ca" />);
    const menu = screen.getByTestId("navbarSupportedContent");
    const button = screen.getByTestId("menuButton");
    expect(menu.classList[1]).toBe("collapse");
    fireEvent.click(button);
    expect(menu.classList[1]).not.toBeDefined();
  });
});
