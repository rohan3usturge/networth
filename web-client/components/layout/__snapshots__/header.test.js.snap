// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly and match snapshot 1`] = `
<div>
  <Head>
    <title>
      Net Worth Calculator
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <span
        className="h4"
      >
        NET WORTH 
        <em>
          calc
        </em>
      </span>
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-testid="menuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      data-testid="navbarSupportedContent"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item "
        >
          <Link
            href="/"
            locale="en-ca"
          >
            <a
              className="nav-link"
            >
              English
            </a>
          </Link>
        </li>
        <li
          className="nav-item "
        >
          <Link
            href="/"
            locale="fr-ca"
          >
            <a
              className="nav-link"
            >
              français
            </a>
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            target="_blank"
          >
            Docs
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`Header should render properly and match snapshot for english 1`] = `
<div>
  <Head>
    <title>
      Net Worth Calculator
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <span
        className="h4"
      >
        NET WORTH 
        <em>
          calc
        </em>
      </span>
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-testid="menuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      data-testid="navbarSupportedContent"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item active"
        >
          <Link
            href="/"
            locale="en-ca"
          >
            <a
              className="nav-link"
            >
              English
            </a>
          </Link>
        </li>
        <li
          className="nav-item "
        >
          <Link
            href="/"
            locale="fr-ca"
          >
            <a
              className="nav-link"
            >
              français
            </a>
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            target="_blank"
          >
            Docs
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`Header should render properly and match snapshot for french 1`] = `
<div>
  <Head>
    <title>
      Net Worth Calculator
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <span
        className="h4"
      >
        NET WORTH 
        <em>
          calc
        </em>
      </span>
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-testid="menuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      data-testid="navbarSupportedContent"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item "
        >
          <Link
            href="/"
            locale="en-ca"
          >
            <a
              className="nav-link"
            >
              English
            </a>
          </Link>
        </li>
        <li
          className="nav-item active"
        >
          <Link
            href="/"
            locale="fr-ca"
          >
            <a
              className="nav-link"
            >
              français
            </a>
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            target="_blank"
          >
            Docs
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
